### DISCUSSION TITLE
引用其他Package创建的reactive对象不能正常实现响应式是正常的吗？

### DISCUSSION THREAD

**Discussion by akinoccc** [vue-mini/vue-mini#95] [Q&A] [answered]
示例
```js
// PackageA 项目
import func from 'packageB'

definePage(() => {
  const data = func() // 返回reactive/ref创建的data

 function change() {
     data.value++; 
 }

  return {
    data,
    change
  }
})
```

```js
// PackageB 独立的npm包
import { ref } from '@vue-mini/core'

export default function() {
  return ref(1)
}
```

貌似是收集不到依赖（dep），所以导致数据改变时，vue-mini监听不到数据的变化，从而没有setData更新
<img width="195" alt="image" src="https://github.com/user-attachments/assets/13cbdc52-b07f-4a8c-bd6a-cf8ba5fd041a">

**Comment by yangmingshan** [comment]
不正常。可能跟你的使用方式有关，需要你提供最小化复现才能调查。

**Reply by akinoccc** [reply]
> 不正常。可能跟你的使用方式有关，需要你提供最小化复现才能调查。

好，我晚点提供一个复现仓库

**Comment by akinoccc** [comment]
@yangmingshan 山哥有空看一下，pnpm dev就可以编译运行了

https://github.com/akinoccc/no-reactive-reproduce.git

**Comment by yangmingshan** [comment, marked as answer]
这个问题本质上是因为 `three-part-package` 引用的 `@vue-mini/core` 和小程序引用的 `@vue-mini/core` 不是同一份。

事实上这里 `three-part-package` 有点问题，`create-vue-mini` 也有问题，两边都需要修改。`three-part-package` 应该把 `@vue-mini/core` 列为 `peerDependencies`，`create-vue-mini` 打包时应该将 `peerDependencies` 视作 `external`

https://github.com/yangmingshan/no-reactive-reproduce/commit/f8af0161614d90687569240f1a9b0c5d4565620a

**Reply by akinoccc** [reply]
> 这个问题本质上是因为 `three-part-package` 引用的 `@vue-mini/core` 和小程序引用的 `@vue-mini/core` 不是同一份。
> 
> 事实上这里 `three-part-package` 有点问题，`create-vue-mini` 也有问题，两边都需要修改。`three-part-package` 应该把 `@vue-mini/core` 列为 `peerDependencies`，`create-vue-mini` 打包时应该将 `peerDependencies` 视作 `external`
> 
> [yangmingshan/no-reactive-reproduce@f8af016](https://github.com/yangmingshan/no-reactive-reproduce/commit/f8af0161614d90687569240f1a9b0c5d4565620a)

👍验证已解决

**Comment by yangmingshan** [comment]
我建了个 issue https://github.com/vue-mini/create-vue-mini/issues/12
